### PR TITLE
First draft of an article about GitHub Actions

### PR DESCRIPTION
**Pull request by davorg**

### CODE REVIEW

File: 2024/incoming/being-lazy-with-github-actions.pod [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          run: |
+            rsync -avz --delete ./ naughtyornice@production-server:/var/www/naughtyornice/
+
+=end code
+
+By adding a secret to the GitHub repository for her server credentials, Pixie
+automated the entire deployment process. The `workflow_dispatch` key in the
+`on` section of the workflow added a "Run" button to the "Actions" tab in the
+GitHub repo, and just pressing that would deploy the application to the
+production server. No more late nights. No more missed sleigh rides. Pixie
```
Comment (openstrike) [inline 2024/incoming/being-lazy-with-github-actions.pod:122]: There are both backticks and double-quotes used here, neither of which are significant in POD. Should one or both be replaced by POD syntax?

**Review by openstrike — APPROVED** [review]
I've not checked the code as I don't use Github Actions, but the prose reads very well.

File: 2024/incoming/being-lazy-with-github-actions.pod (hunk 1 above)
Comment (davorg) [reply 2024/incoming/being-lazy-with-github-actions.pod]: Oops. Yes. The problem of switching between Pod and Markdown. Thanks for spotting it. Fixed.

File: 2024/incoming/being-lazy-with-github-actions.pod [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+it was for the tech-savvy elves in charge of JavaScript or Rust projects.
+The article made it seem that GitHub Actions were just as useful for Perl
+developers, and that it could automate much of her workflow.
+
+Intrigued, Pixie decided to dive in.
+
+=head2 A First Workflow: Automating Tests
+
+Pixie started with the basics: automating the test suite for NaughtyOrNice.
+The app already had a robust set of tests written with Test::More, but running
```
Comment (oalders) [inline 2024/incoming/being-lazy-with-github-actions.pod:45]: Should we link to Test::More?

File: 2024/incoming/being-lazy-with-github-actions.pod [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        uses: PerlToolsTeam/github_workflows/.github/workflows/cpan-perlcritic.yml@main
+
+      complexity:
+        uses: PerlToolsTeam/github_workflows/.github/workflows/cpan-complexity.yml@main
+
+=end code
+
+Pixie was very happy. In a few lines of code, she had replaced and improved
+on her existing manual processes. The application was now being tested
+automatically on several versions of Perl and on Linux, Windows and MacOS. She
```
Comment (oalders) [inline 2024/incoming/being-lazy-with-github-actions.pod:168]: I think the latest marketing has it as `macOS` rather than `MacOS`.

**Review by oalders — COMMENTED** [review]
This is very helpful. I didn't know you could include 3rd party workflows like this.

File: 2024/incoming/being-lazy-with-github-actions.pod [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+          - main
+      pull_request:
+
+    jobs:
+      test:
+        runs-on: ubuntu-latest
+
+        steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
```
Comment (JJ) [inline 2024/incoming/being-lazy-with-github-actions.pod:66]: Not the latest version any more.

File: 2024/incoming/being-lazy-with-github-actions.pod [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+        - name: Set up Perl
+          uses: shogo82148/actions-setup-perl@v1
+          with:
+            perl-version: '5.40'
+
+        - name: Install dependencies
+          run: |
```
Comment (JJ) [inline 2024/incoming/being-lazy-with-github-actions.pod:74]: You don't really need this for a single line

**Review by JJ — COMMENTED** [review]
Nice overall!

**Comment by oalders** [comment]
Thanks @davorg! 🎄

**Pull request merged**